Add HTML report

sci-kai opened the pull request:
# Add HTML report using datavzrd

This PR adds the generation of an HTML report based on the TSV file created by vembrane.
This report will be the main output of the workflow and should help in interpreting the variants

## Features

- __New local module datavzrd.__
- __Generation of an excel file__ with datavzrd that can be accessed through the HTMl report.
- __Splitting of the variant TSV file columns based on predefined groups__ using a python script [preprocess_datavzrd.py](bin/preprocess_datavzrd.py). These generates separate tables, that are linked with each other in the HTML report and are separate separate tabs in the excel file.
- __a datavzrd_config_template can automatically render tick plots, heatmaps and hyperlinks__ based on definitions in the [annotation_colinfo.tsv](assets/annotation_colinfo.tsv) file. The yaml file is rendered with the YTE engine and has access to the columns from the annotation_colinfo.tsv file.
In general, this makes the adaptation, extension and design changes for the HTML report much more easy and adaptable through simple modifications of the annotation_colinfo.tsv file. Please have a look at the help text of the new parameters which describe these two new assets in detail. 
- __A draft for a HTML report design__ that is defined in the annotation_colinfo.tsv file. This should be discussed together and need to be extended and adapted based on design decisions.
- added an __identifier for each variant__ that is based on the columns: "chrom","pos","ref","alt","id","Feature". The latter is added as the TSV file is per variant-transcript. This is used to reference between tables and define unique entries.
- new parameters:
    - report: boolean for generating report
    - datavzrd_config: optional, needs a template that is provided in [assets/datavzrd_config_template.yaml](assets/datavzrd_config_template.yaml)
    - annotation_colinfo: optional, needs a annotation overview that is provided in [assets/annotation_colinfo.tsv](assets/annotation_colinfo.tsv)

## To-Do's:

- It still needs to define docker and singularity containers with datavzrd and the necessary python packages for easy usage.
- refining the HTML report design: adding links, splitting and formatting columns as e.g. dbSNP entries, SIFT column, etc.

biolancer reviewed (approved):
Looks good to me, we will probably update the module in multiple iterations either way depending on the output files we create. The HTML report looks good, so for me: LGTM! Thanks a lot Kai!

sci-kai commented:
I fixed a bug with channel handling. 
Beforehand, it only generated one report for one sample, now the channels are combined so that each sample generates a report.
I also changed the process resource requirements for `vembrane table` and `datavzrd` to more realistic requirements.

feiloo reviewed (approved):
Looks good
